src/jdk.javadoc/share/classes/jdk/javadoc/internal/doclets/toolkit/taglets/ParamTaglet.java

```diff
@@ -62,21 +62,19 @@ public ParamTaglet() {
     }
 
     /**
-     * Given a list of parameters, return a name/rank number map.
-     * If the list is null, then null is returned.
-     * @param params The list of parameters (from type or executable member) to
-     *               check.
-     * @return a name-rank number map.
+     * Given a list of parameters, returns a name-position map.
+     * @param params the list of parameters from a type or an executable member
+     * @return a name-position map
      */
-    private static Map<String, String> getRankMap(Utils utils, List<? extends Element> params) {
-        HashMap<String, String> result = new HashMap<>();
-        int rank = 0;
+    private static Map<String, String> mapNameToPosition(Utils utils, List<? extends Element> params) {
+        Map<String, String> result = new HashMap<>();
```

Comment:
We could change it to `Integer` now, but it wouldn't help us much: the `DocFinder.Input.tagId` field, which that `Integer` is ultimately compared against, is of type `String`. To make it all the way, we would need to change `tagId` to `Integer` too. But that won't work because `tagId` is also used by `ThrowsTaglet` to store an exception class (or an exception type variable) name.

That said, I agree that we should change `String` to `Integer` there. Let's just not do it now as it requires deeper refactoring; I have a plan already and some implementation in the works.

Reply:
Noted about tagId. You could still do a local cleanup for the type of the map, but I guess we'll wait and see what your plan is.

Reply:
> Noted about tagId. You could still do a local cleanup for the type of the map, but I guess we'll wait and see what your plan is.

I cleaned up locally in 453ecbe. Mind you, the "deeper refactoring" is tracked by JDK-8285488.